File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  const map = {};
  for (let i = 0; i < nums.length; i++) {
    const num = nums[i];
    const neededNumber = target - num;
    if (map[neededNumber] !== undefined) {
      return [map[neededNumber], i];
    }
    map[num] = i;
  }
  return [];
};

File: 0001-two-sum/0001-two-sum.cpp
class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        unordered_map<int, int> map;
        for (int i = 0; i < nums.size(); i++)
        {
            int curr = nums[i];
            int needed = target - curr;

            if (map.find(needed) != map.end())
            {
                return {i, map[needed]};
            }
            map[curr] = i;
        }
        return {};
    }
};